src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalNamedFormulaRule.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.App.Controls;
+using Microsoft.PowerFx.Core.Lexer.Tokens;
+using Microsoft.PowerFx.Core.Syntax;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Core.App
+{
+    /// <summary>
+    /// Wraps a named formula's Formula, Rule, Token, and ResultType.
+    /// </summary>
+    internal interface IExternalNamedFormulaRule
```

Comment:
>IExternalNamedFormulaRule [](http://example.com/codeflow?start=23&length=25)

We should be removing the Externals to reduce coupling. 

Do we need the whole structure here? Or just Async?
If we do need it, is there a first class way to do it in power fx.

Reply:
we just need to know if the rule is async.